frontend/templates/product-list/hooks/useVariantProductList.ts

```diff
@@ -0,0 +1,43 @@
+import { ProductList } from '@models/product-list';
+import {
+   destylizeDeviceTitle,
+   destylizeDeviceTitleAndVariant,
+   splitDeviceAndVariant,
+} from '@helpers/product-list-helpers';
+
+export function useVariantProductList(
+   productList: ProductList,
+   deviceHandle?: string
+) {
+   if (!deviceHandle) return productList;
+   const { variant } = splitDeviceAndVariant(deviceHandle);
+   if (!variant) return productList;
+
+   const modelNumberPattern = productList.wikiInfo?.find(
+      (info) => info.name === 'model_number_pattern'
+   )?.value;
+   const modelNumberRegex = modelNumberPattern
+      ? new RegExp(modelNumberPattern?.replace(/[*⎵]/g, '.'), 'g')
+      : null;
+   const replaceModelNumber =
+      modelNumberRegex && variant.match(modelNumberRegex);
```

Comment:
I don't think we need to match the variant against the pattern. We already know the variants belong to the device which is enough. We just need to do the replacement i.e. `deviceDisplayTitle.replaceAll(modelNumberPattern, variant)` and if the pattern doesn't exist in the deviceTitle then we do `deviceDisplayTitle variant` with a space. This should be used everywhere that the device title gets used (including itemType pages).

Reply:
So to be clear, whenever a device title contains a model number pattern in it **all** of its variants will match that pattern? 

What about if someone types in the variant in their url? It should show an inaccurate replace because its nonsense anyway?

> pattern doesn't exist in the deviceTitle then we do deviceDisplayTitle variant with a space

there is no point to the pattern then. lets just delete those

Reply:
I agree that we don't need to use model number patterns at all in Next. We should use the variant from the url no matter what.

> What about if someone types in the variant in their url? It should show an inaccurate display title because the request is nonsense?

Yes, that's what we do for the item type suffix. The only way a user could end up in this state is if they url hack

https://www.ifixit.com/Parts/Phone/asdf

![image](https://github.com/iFixit/react-commerce/assets/52104630/8d881536-9b76-4b1e-ab3e-fe5fc104d45d)

Reply:
I still personally feel like the regex is the cleanest solution that is resilient to data mistakes IF we are using model number pattern matching